learn/protocol/block-format.md

```diff
@@ -1,95 +1,112 @@
-| `extra_data` | Field reserved for arbitrary data. |
-| `state_root` | The root of the Merkle tree representing the blockchain state, acting as a commitment to its current state. |
-| `diff_root` | The root of the trie diff tree proof, authenticating the output of the RequestChunk. |
-| `body_root` | The root of the Merkle tree representing the body of the block, serving as a commitment to its content. |
-| `history_root` | The root of a Merkle Mountain Range covering all transactions in the current epoch. |
+| **Field** | **Data Type** | **Description** |
+| --- | --- | --- |
+| `network` | `NetworkId` | The network ID associated with the block, such as Mainnet or Testnet |
+| `version` | `u16` | The block's version number. A change in the version number implies a hard fork. It can currently only be 1. |
+| `block_number` | `u32` | The number of the block, representing its height in the blockchain |
+| `timestamp` | `u64` | The block's Unix creation timestamp in milliseconds, indicating when the block was produced |
+| `parent_hash` | `Blake2bHash` | The hash of the preceding block's header (micro or macro). This ensures a direct link to its predecessor |
+| `seed` | `VrfSeed` | The BLS signature derived from the seed of the previous block, using the validator key of the block producer |
+| `extra_data` | `Vec<u8>` | Data that can be freely chosen by the producing validator, the default client leaves it empty |
+| `state_root` | `Blake2bHash` | The root of the Merkle tree representing the blockchain state, acting as a commitment to the current state |
+| `body_root` | `Blake2sHash` | The hash of the block's body, serving as a commitment to its content |
+| `diff_root` | `Blake2bHash` | The root of the trie diff tree proof, authenticating the state changes between blocks |
+| `history_root` | `Blake2bHash` | The root of a Merkle Mountain Range covering all transactions in the current epoch up until the current block|
 
-### Micro body
+### Micro Body
 
-| Data field | Description |
-| --- | --- |
-| `equivocation_proofs` | Equivocation proofs for this block. This field could be empty as equivocation proofs aren't present in every block. |
-| `transactions` | Contains all the transactions of the block. This field might be empty in blocks without any transactions. (see [skip blocks](/learn/protocol/validators/skip-blocks.md)). |
+| **Field** | **Data Type** | **Description** |
+| --- | --- | --- |
+| `equivocation_proofs` | `Vec<EquivocationProof>` | A vector containing equivocation proofs for this block. This field may be empty if no such proofs exist in the block |
+| `transactions` | `Vec<ExecutedTransaction>` | A vector containing the transactions for this block. It may be empty in blocks without any transactions, as in skip blocks |
```

Comment:
Both these fields are also empty if there are simply no (equivocation proofs or) transactions that could potentially be included in the block. The producer can freely choose which available transactions to include in its produced block.